### PR TITLE
Remove test data from esmpy package

### PR DESCRIPTION
**Pull request by xylar**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by xylar** [comment]
@conda-forge-admin, please rerender

**Comment by xylar** [comment]
I'll continue with this once #81 is resolved.

**Comment by xylar** [comment]
This worked great in a local test build!  The package is now only 2 MB in size.

File: recipe/build.sh [hunk 1]
```diff
@@ -13,6 +13,7 @@ ${PYTHON} -m pip install .
 if [[ "$CONDA_BUILD_CROSS_COMPILATION" != "1" ]]; then
     export ESMF_MPIRUN=${SRC_DIR}/src/Infrastructure/stubs/mpiuni/mpirun
     make test_serial
+    rm -rf $SP_DIR/esmpy/data
```
Comment (xylar) [inline recipe/build.sh:16]: @billsacks, @zklaus, @ocefpaf, any concerns about this?

**Comment by xylar** [comment]
If no one has concerns, please merge as soon as CI has passed.

**Comment by xylar** [comment]
I think I'm just going to take the chance and merge this.  I think it works and the data isn't something we want to be shipping.  Hopefully, you all agree.

File: recipe/build.sh (hunk 1 above)
Comment (zklaus) [reply recipe/build.sh:16]: Looks ok to me.

**Review by zklaus — APPROVED** [review]

**Comment by zklaus** [comment]
I thought there might be a risk that there was non-test data in the directory, but I confirmed that this is not the case (currently). I also thought that the directory might need to be present even if we remove its contents, but I confirmed that the download routines create the directory if it is missing.

**Comment by zklaus** [comment]
Thanks, @xylar and @mx-moth for reporting it. Great improvement of the package.

File: recipe/build.sh (hunk 1 above)
Comment (ocefpaf) [reply recipe/build.sh:16]: Thanks for doing this!

**Comment by xylar** [comment]
I would still prefer to use the option of `${ESMPY_DATA_DIR}` mentioned in #78 but I couldn't get that to work, and I don't know if that's just because I don't understand it or because it's really broken.  I'll revisit when I hear back from the ESMF folks.

**Comment by billsacks** [comment]
I spent some time looking into this today.

First, I agree that the change in this PR (removing the data directory) is a fine workaround for now: I agree that it looks like this is just used by tests and examples, so shouldn't be needed in the esmpy installation.

`ESMPY_DATA_DIR` is used in some of our nightly testing to avoid potentially long download times when we can instead reuse already-existing data. As you found, the current assumption in the code is that, if `ESMPY_DATA_DIR` is defined, then it is already populated (e.g., by doing a git clone of our test data repo) and so we should save time by not trying to download data. This is a feature (though one that may not be widely useful) rather than a bug, but I also agree that your use case is worth supporting.

My proposal is to add a new environment variable (I suggest `ESMPY_DATA_NEW_DIR`, but I'm open to suggestions) that has the behavior that I think you expect: If this new variable is defined (and `ESMPY_DATA_DIR` is not defined) then that location will be used for the data download rather than the default location. From looking through the code, I think this should be straightforward to implement, so I'll move ahead with that if it sounds good to you.

**Comment by xylar** [comment]
@billsacks, I very much like your suggested solution and variable name.  That sounds great to me!

**Comment by billsacks** [comment]
I have implemented this in the latest ESMF develop (https://github.com/esmf-org/esmf/commit/cb4b2065ee06180ffac9314d92036778a1132aab). It will be available in the ESMF 8.6 release due out this fall.

**Comment by xylar** [comment]
Excellent, thank you @billsacks!

**Pull request merged**